xonsh/xoreutils/_which.py

```diff
@@ -191,21 +192,21 @@ def whichgen(command, path=None, verbose=0, exts=None):
     # Windows has the concept of a list of extensions (PATHEXT env var).
     if sys.platform.startswith("win"):
         if exts is None:
-            exts = os.environ.get("PATHEXT", "").split(os.pathsep)
+            exts = builtins.__xonsh_env__['PATHEXT']
```

Comment:
If it needs changing (and it sounds like it does), maybe we can leave a standard comment/note on any lines that are xonsh-specific for anyone who wants to use it outside of xonsh?